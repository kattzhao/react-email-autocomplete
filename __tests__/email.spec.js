import React from 'react'
import { shallow, mount } from 'enzyme'

import Email from '../src/index'

const customDomains = ['jest.com', 'enzyme.com']

describe('Email Component', () => {
  it('should render correctly without props', () => {
    const component = shallow(<Email />)
    expect(component).toMatchSnapshot()
  })
  it('should suggest test@gmail.com', () => {
    const component = mount(<Email />)

    // enter an email address
    component
      .find('input')
      .simulate('change', { target: { value: 'test@gm' } })

    expect(component.instance().state.value).toEqual('test@gmail.com')
  })
  it('should suggest test@jest.com', () => {
    const component = mount(<Email domains={customDomains} />)

    // enter an email address
    component
      .find('input')
      .simulate('change', { target: { value: 'test@je' } })

    expect(component.instance().state.value).toEqual('test@jest.com')
  })
})
