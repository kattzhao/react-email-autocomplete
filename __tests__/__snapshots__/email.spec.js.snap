// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Component should render correctly without props 1`] = `
<div
  className="rea-wrapper"
>
  <input
    autoCapitalize="none"
    domains={
      Array [
        "yahoo.com",
        "hotmail.com",
        "gmail.com",
        "me.com",
        "aol.com",
        "mac.com",
        "live.com",
        "googlemail.com",
        "msn.com",
        "facebook.com",
        "verizon.net",
        "outlook.com",
        "icloud.com",
        "rediffmail.com",
        "zohomail.in",
        "zoho.com",
        "mail.com",
        "protonmail.com",
        "yandex.com",
      ]
    }
    inputMode="email"
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value=""
  />
</div>
`;
